packages/plugins/documentation/server/services/documentation.js

```diff
@@ -2,49 +2,69 @@
 
 const path = require('path');
 const fs = require('fs-extra');
-const _ = require('lodash');
+const { cloneDeep } = require('lodash/fp');
 const { getAbsoluteServerUrl } = require('@strapi/utils');
-
 const { builApiEndpointPath, buildComponentSchema } = require('./helpers');
 
 const defaultOpenApiComponents = require('./utils/default-openapi-components');
+const { getPluginsThatNeedDocumentation } = require('./utils/get-plugins-that-need-documentation');
+
+/**
+ * @description
+ * Mutates the current state of the OpenAPI object and returns a new immutable object
+ *
+ * @param {object} currentState The immutable state of the OpenAPI object
+ * @param {function} mutateStateCallback A callback function that can mutate a copy of the immutable state
+ *
+ * @returns {object}
+ * The mutated copy as the new immutable state
+ */
+const mutateDocumentation = (currentState, mutateStateCallback) => {
+  // Create a copy of the current state that is mutable
+  const draftState = cloneDeep(currentState);
+  // Pass the draft to the callback for mutation
+  mutateStateCallback(draftState);
+  // Return the mutated state as a new immutable state
+  return Object.freeze(draftState);
+};
```

Comment:
ℹ️ Maybe this is overkill and it is fine to just mutate the object directly 🤷‍♂️ , but I liked the idea of the current state of the generated documentation always being immutable outside this scope

Reply:
Could we also try using immer to simplify the immutable logic?

Reply:
I had immer before but then I started wondering if the package was really necessary since a similar result could be accomplished with this simple function. How do you think the logic would be more simple with immer?

Reply:
if it was just for this then it would be overkill, but since it's used by the admin package, I don't think (but to be confirmed) it would increase the node modules. It's just if we can get it for free here that it would be nice to simplify this logic, otherwise no problem with the current implementation